src/main/java/org/gridsuite/study/server/StudyService.java

```diff
@@ -67,106 +65,93 @@ public StudyService(
             @Value("${backing-services.geo-data.base-uri:http://geo-data-store-server/}") String geoDataServerBaseUri,
             @Value("${backing-services.network-map.base-uri:http://network-map-store-server/}") String networkMapServerBaseUri,
             @Value("${backing-services.network-modification.base-uri:http://network-modification-server/}") String networkModificationServerBaseUri,
-            @Value("${backing-services.loadflow.base-uri:http://loadflow-server/}") String loadFlowServerBaseUri) {
-        RestTemplateBuilder restTemplateBuilder = new RestTemplateBuilder();
-        caseServerRest = restTemplateBuilder.build();
-        caseServerRest.setUriTemplateHandler(new DefaultUriBuilderFactory(caseServerBaseUri));
-
-        singleLineDiagramServerRest = restTemplateBuilder.build();
-        singleLineDiagramServerRest.setUriTemplateHandler(new DefaultUriBuilderFactory(singleLineDiagramServerBaseUri));
-
-        networkConversionServerRest = restTemplateBuilder.build();
-        networkConversionServerRest.setUriTemplateHandler(new DefaultUriBuilderFactory(networkConversionServerBaseUri));
-
-        geoDataServerRest = restTemplateBuilder.build();
-        geoDataServerRest.setUriTemplateHandler(new DefaultUriBuilderFactory(geoDataServerBaseUri));
-
-        networkMapServerRest = restTemplateBuilder.build();
-        networkMapServerRest.setUriTemplateHandler(new DefaultUriBuilderFactory(networkMapServerBaseUri));
-
-        networkModificationServerRest = restTemplateBuilder.build();
-        networkModificationServerRest.setUriTemplateHandler(new DefaultUriBuilderFactory(networkModificationServerBaseUri));
-
-        loadFlowServerRest = restTemplateBuilder.build();
-        loadFlowServerRest.setUriTemplateHandler(new DefaultUriBuilderFactory(loadFlowServerBaseUri));
+            @Value("${backing-services.loadflow.base-uri:http://loadflow-server/}") String loadFlowServerBaseUri,
+            NetworkStoreService networkStoreClient,
+            StudyRepository studyRepository) {
+        this.caseServerBaseUri = caseServerBaseUri;
+        this.singleLineDiagramServerBaseUri = singleLineDiagramServerBaseUri;
+        this.networkConversionServerBaseUri = networkConversionServerBaseUri;
+        this.geoDataServerBaseUri = geoDataServerBaseUri;
+        this.networkMapServerBaseUri = networkMapServerBaseUri;
+        this.networkModificationServerBaseUri = networkModificationServerBaseUri;
+        this.loadFlowServerBaseUri = loadFlowServerBaseUri;
+
+        ExchangeStrategies exchangeStrategies = ExchangeStrategies.builder()
+                .codecs(configurer -> configurer.defaultCodecs().maxInMemorySize(1024 * 1024 * 10)).build();
```

Comment:
Why do we need ```maxInMemorySize(1024 * 1024 * 10)```

Reply:
If not the buffering limits on the webflux codecs will be exceeded.

Reply:
What are the limits? What happens when they are exceeded ? In what case do we exceed them ?